instrumentation/spring/spring-webmvc-3.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/springwebmvc/SpringWebMvcTracer.java

```diff
@@ -36,7 +36,12 @@ public void onRequest(Span span, HttpServletRequest request) {
       Object bestMatchingPattern =
           request.getAttribute(HandlerMapping.BEST_MATCHING_PATTERN_ATTRIBUTE);
       if (bestMatchingPattern != null) {
-        span.updateName(bestMatchingPattern.toString());
+        String contextPath = request.getContextPath();
+        if (contextPath != null && !contextPath.isEmpty() && !contextPath.equals("/")) {
+          span.updateName(contextPath + bestMatchingPattern.toString());
```

Comment:
This is essentially one more way to calculate span name. Why we don't use `BaseTracer.getApplicationRoot` here? We have similar but something different rule scattered across the code base.

Reply:
the latest commit consolidates everything to single approach